### ISSUE TITLE
get overfit after severl epochs

### ISSUE THREAD

**Issue by zhangnn016** [experiencor/basic-yolo-keras#187]
i use the raccoon dataset, and warm up the net for 3 epochs. when i train it with tiny yolo or squeezenet backend, the test loss always decreases but the val loss increases after about 20 epochs, why does this happens? please help me, thank you!

**Comment by zhangnn016** [comment]
sorry, i mean the train loss decreases but the val loss increases after about 20 epochs, and stays at about val_loss=12.0

**Comment by experiencor** [comment]
Hi @zhangnn016! It's a little odd that val_loss stays that high. Did you ```Set pretrained_weights setting in config.json to the warmup weights (whatever in saved_weights_name)```. 

Other than that, it's a bit tricky with SqueezeNet and TinyYolo backends. I had to play with the 4 scale settings to get them to work. But FullYolo will work out of the box.

**Comment by zhangnn016** [comment]
Hi @experiencor, Thanks for your reply!  Because I want to use the detection on mobile devices so I have to choose simple net, such as mobilenet, tiny-yolo and squeezenet. I'm sure I have done the `Set pretrained_weights setting in config.json to the warmup weights (whatever in saved_weights_name)`, but I didn't use the pretrained weights you provided but start from scratch. May be I should your pretrained weights, I will try it, could you give me some advices about the 4 scale settings? Thanks you very much!!

**Comment by zeynali** [comment]
@zhangnn016 , Please put the config.json file here.

**Comment by zhangnn016** [comment]
@zeynali , @experiencor, this is the original weights：
`{
    "model" : {
        "architecture":         "SqueezeNet",
        "input_size":           416,
        "anchors":              [1.75,2.21, 3.49,5.33, 4.24,10.19, 6.51,6.69, 8.91,9.82],
        "max_box_per_image":    10,
        "labels":               ["kangaroo"]
    },

    "train": {
        "train_image_folder":   "/Develop/DataSet/kangaroo/images/",
        "train_annot_folder":   "/Develop/DataSet/kangaroo/annots/",
          
        "train_times":          10,
        "pretrained_weights":   "",
        "batch_size":           16,
        "learning_rate":        1e-4,
        "nb_epoch":             50,
        "warmup_epochs":        3,

        "object_scale":         5.0 ,
        "no_object_scale":      0.5,
        "coord_scale":          1.0,
        "class_scale":          1.0,

        "saved_weights_name":   "squeezenet_kangaroo.h5",
        "debug":                true
    },

    "valid": {
        "valid_image_folder":   "",
        "valid_annot_folder":   "",

        "valid_times":          1
    }
}
`
after warm up, i changed config.json to this:
`{
    "model" : {
        "architecture":         "SqueezeNet",
        "input_size":           416,
        "anchors":              [1.75,2.21, 3.49,5.33, 4.24,10.19, 6.51,6.69, 8.91,9.82],
        "max_box_per_image":    10,
        "labels":               ["kangaroo"]
    },

    "train": {
        "train_image_folder":   "/Develop/DataSet/kangaroo/images/",
        "train_annot_folder":   "/Develop/DataSet/kangaroo/annots/",
          
        "train_times":          10,
        "pretrained_weights":   "squeezenet_kangaroo.h5",
        "batch_size":           16,
        "learning_rate":        1e-4,
        "nb_epoch":             50,
        "warmup_epochs":        0,

        "object_scale":         5.0 ,
        "no_object_scale":      0.5,
        "coord_scale":          1.0,
        "class_scale":          1.0,

        "saved_weights_name":   "squeezenet_kangaroo_2.h5",
        "debug":                true
    },

    "valid": {
        "valid_image_folder":   "",
        "valid_annot_folder":   "",

        "valid_times":          1
    }
}
`

when trainning, the loss_WH is always much higher than loss_XY and loss_Class, such as:
`60/90 [===================>..........] - ETA: 1:12 - loss: 7.3641Dummy Line 	[0]
Loss XY 	[0.0669294074]
Loss WH 	[7.73973036]
Loss Conf 	[0.000757656293]
Loss Class 	[0]
Total Loss 	[7.80741739]
Current Recall 	[0]
Average Recall 	[0]
`
the val_loss is also very high:

` 214s 2s/step - loss: 8.3809 - val_loss: 10.9439`

please help me, thank you!

**Comment by zeynali** [comment]
@zhangnn016 , i have some problem , Apparently this codes have problem , everyone trained this code , all reach 0 recall , Unfortunately , sir @experiencor , don't attention with these problem , And to everyone they say, please, take a look at more detail in Section 2/4. If he has the time to solve the problem, will be fine.

**Comment by zhangnn016** [comment]
@zeynali  I try full-yolo model in the config.json I wrote above, the val_loss is ok and it can detect kangaroo correctly,  may be the code only run on full-yolo?  You can have a try!

**Comment by zeynali** [comment]
@zhangnn016 please put your config.json here.
